core/arch/arm/plat-stm32mp1/main.c

```diff
@@ -247,64 +245,128 @@ void boot_secondary_init_intc(void)
+				      CFG_STM32MP1_SCMI_SHM_SIZE,
+				      base, size))
+		return TEE_SUCCESS;
+
+	res = firewall_set_memory_configuration(&fw_query, base, size);
+	if (res)
+		EMSG("Failed to configure non-secure SRAM %#"PRIxPA"..%#"PRIxPA,
+		     base, base + size);
+
+	return res;
+}
 
-#if TZSRAM_FITS_IN_SYSRAM_AND_SRAMS || TZSRAM_FITS_IN_SRAMS || \
-	SCMI_SHM_IS_IN_SRAMX
 /* At run time we enforce that SRAM1 to SRAM4 are properly assigned if used */
-static TEE_Result init_stm32mp15_secure_srams(void)
+static void configure_srams(struct dt_driver_provider *fw_provider)
 {
+	bool error = false;
+
 	if (IS_ENABLED(CFG_WITH_PAGER)) {
-		if (core_is_buffer_intersect(CFG_TZSRAM_START, CFG_TZSRAM_SIZE,
-					     SRAM1_BASE, SRAM1_SIZE))
-			stm32mp_register_secure_periph_iomem(SRAM1_BASE);
+		if (secure_pager_ram(fw_provider, STM32MP1_ETZPC_SRAM1_ID,
+				     SRAM1_BASE, SRAM1_SIZE))
+			error = true;
 
-		if (core_is_buffer_intersect(CFG_TZSRAM_START, CFG_TZSRAM_SIZE,
-					     SRAM2_BASE, SRAM2_SIZE))
-			stm32mp_register_secure_periph_iomem(SRAM2_BASE);
+		if (secure_pager_ram(fw_provider, STM32MP1_ETZPC_SRAM2_ID,
+				     SRAM2_BASE, SRAM2_SIZE))
+			error = true;
 
-		if (core_is_buffer_intersect(CFG_TZSRAM_START, CFG_TZSRAM_SIZE,
-					     SRAM3_BASE, SRAM3_SIZE))
-			stm32mp_register_secure_periph_iomem(SRAM3_BASE);
+		if (secure_pager_ram(fw_provider, STM32MP1_ETZPC_SRAM3_ID,
+				     SRAM3_BASE, SRAM3_SIZE))
+			error = true;
 
-		if (core_is_buffer_intersect(CFG_TZSRAM_START, CFG_TZSRAM_SIZE,
-					     SRAM4_BASE, SRAM4_SIZE))
-			stm32mp_register_secure_periph_iomem(SRAM4_BASE);
+#ifdef CFG_STM32MP15
+		if (secure_pager_ram(fw_provider, STM32MP1_ETZPC_SRAM4_ID,
+				     SRAM4_BASE, SRAM4_SIZE))
+			error = true;
+#endif
 	}
+	if (CFG_STM32MP1_SCMI_SHM_BASE) {
+		if (non_secure_scmi_ram(fw_provider, STM32MP1_ETZPC_SRAM1_ID,
+					SRAM1_BASE, SRAM1_SIZE))
+			error = true;
+
+		if (non_secure_scmi_ram(fw_provider, STM32MP1_ETZPC_SRAM2_ID,
+					SRAM2_BASE, SRAM2_SIZE))
+			error = true;
+
+		if (non_secure_scmi_ram(fw_provider, STM32MP1_ETZPC_SRAM3_ID,
+					SRAM3_BASE, SRAM3_SIZE))
+			error = true;
 
-	if (SCMI_SHM_IS_IN_SRAMX) {
-		if (core_is_buffer_intersect(CFG_STM32MP1_SCMI_SHM_BASE,
-					     CFG_STM32MP1_SCMI_SHM_SIZE,
-					     SRAM1_BASE, SRAM1_SIZE))
-			stm32mp_register_non_secure_periph_iomem(SRAM1_BASE);
-
-		if (core_is_buffer_intersect(CFG_STM32MP1_SCMI_SHM_BASE,
-					     CFG_STM32MP1_SCMI_SHM_SIZE,
-					     SRAM2_BASE, SRAM2_SIZE))
-			stm32mp_register_non_secure_periph_iomem(SRAM2_BASE);
-
-		if (core_is_buffer_intersect(CFG_STM32MP1_SCMI_SHM_BASE,
-					     CFG_STM32MP1_SCMI_SHM_SIZE,
-					     SRAM3_BASE, SRAM3_SIZE))
-			stm32mp_register_non_secure_periph_iomem(SRAM3_BASE);
-
-		if (core_is_buffer_intersect(CFG_STM32MP1_SCMI_SHM_BASE,
-					     CFG_STM32MP1_SCMI_SHM_SIZE,
-					     SRAM4_BASE, SRAM4_SIZE))
-			stm32mp_register_non_secure_periph_iomem(SRAM4_BASE);
+#ifdef CFG_STM32MP15
```

Comment:
I may be in favor to globalize the use of `#if defined(xxx)` in our files

Reply:
ok